pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -377,16 +377,12 @@ func (dswp *desiredStateOfWorldPopulator) checkVolumeFSResize(
 		// or online resize in subsequent loop(after we confirm it has been mounted).
 		return
 	}
-	fsVolume, err := util.CheckVolumeModeFilesystem(volumeSpec)
+	_, err := util.CheckVolumeModeFilesystem(volumeSpec)
 	if err != nil {
 		klog.Errorf("Check volume mode failed for volume %s(OuterVolumeSpecName %s): %v",
 			uniqueVolumeName, podVolume.Name, err)
 		return
 	}
-	if !fsVolume {
-		klog.V(5).Infof("Block mode volume needn't to check file system resize request")
-		return
-	}
 	if processedVolumesForFSResize.Has(string(uniqueVolumeName)) {
```

Comment:
Just to be on safe side, for in-tree drivers I think we should still skip calling fs resize on the node if volume is of type raw block. 

The other thing is - we are going to need some e2e tests for this.

Reply:
What if migration is enabled?

in-tree: no Pending condition on PVC if mode Block
CSI: always Pending condition on PVC
migrated CSI: ?

Reply:
Our e2e test expects that Block volumes don't get the condition
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volume_expand.go#L190
I'm running the tests for aws migration.
So the test needs to know if the plugin is CSI or in-tree or migrated CSI which will be ugly.

Reply:
That e2e is broken for CSI raw block volumes, no matter what we do here right? Plugin already knows if it is CSI or in-tree. I guess in-tree migrated CSI is tricky. But I think this ugliness will go away once we merge - https://github.com/container-storage-interface/spec/pull/381 

So our options are:
1. Keep in-tree drivers behaviour same as CSI, but since in CSI too we are moving towards skipping `NodeExpandVolume` for raw block devices, this seems like a step backward and will cause unnecessary churn.
2. Live with ugliness in e2e until we https://github.com/container-storage-interface/spec/pull/381 is merged and fix the in-tree <-> csi drivers.

I am leaning towards #2. And yeah - I think we should not skip calling fs resize if driver is in-tree but migration is enabled for it.

Reply:
I have opened https://github.com/kubernetes/kubernetes/pull/81611 to fix this problem. I think the condition checking there added very little value.